src/node_file-inl.h

```diff
@@ -349,6 +349,38 @@ int SyncCall(Environment* env, v8::Local<v8::Value> ctx,
   return err;
 }
 
+// Similar to SyncCall but throws immediately if there is an error.
+template <typename Predicate, typename Func, typename... Args>
+int SyncCallAndThrowIf(Predicate should_throw,
+                       Environment* env,
+                       FSReqWrapSync* req_wrap,
+                       Func fn,
+                       Args... args) {
+  env->PrintSyncTrace();
+  int result = fn(env->event_loop(), &(req_wrap->req), args..., nullptr);
```

Comment:
Do we need env->event_loop for sync call that doesn't have a callback?

Reply:
I don't actually know for sure in the case of sync fs methods, but we've been doing this ever since libuv changed the signature of these methods to take an event loop: https://github.com/nodejs/node/commit/21cc4c4985fdf40cc128cd88cfdc02fd49e33b26 maybe @nodejs/libuv would know.

Reply:
As far as I know it's not necessary. uvwasi passes `NULL` for all of its sync calls.

Reply:
Current `sync` implementations on `main` branch pass `nullptr` as event_loop parameter, and passes all tests.

Reply:
Switched to nullptr. If that turns out to be problematic for whatever reason, we can switch back